gadget/gadget.go

```diff
@@ -577,8 +597,18 @@ func validateRole(vs *VolumeStructure, vol *Volume) error {
 
 	switch vsRole {
 	case SystemData:
-		if vs.Label != "" && vs.Label != ImplicitSystemDataLabel {
-			return fmt.Errorf(`role of this kind must have an implicit label or %q, not %q`, ImplicitSystemDataLabel, vs.Label)
+		dataLabel := ImplicitSystemDataLabel
+		if constraints == nil {
+			if state != nil && state.SystemSeed {
```

Comment:
This feels like we'll get inconsistent handling when system-seed is defined after system-data structure.

Also, this feels like a really voulme-level check, perhaps better suited in `validateVolume()`, where you wouldn't really need to pass the state information around.

Reply:
Indeed it requires `system-seed` to be defined before `system-data`, but the laid out volume is ordered and `system-data` is the last structure so it can be growable. But in the general case I think you're right, we can collect data and check overall consistency in the end.

Reply:
Changed code to do consistency check at the end.